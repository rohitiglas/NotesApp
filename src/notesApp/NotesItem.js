import React, {memo} from 'react';
import {View, Text, Button} from 'react-native';
const NotesItem = ({note, getNote, deleteNote}) => {
  console.log("LSLLSLSLSLLLLSLNote",note);
  return (
    <View style={{margin: 10}}>
      <Text testID={'title'} style={{fontSize: 22, fontWeight: 'bold'}}>{note.title}</Text>
      <Text testID={'description'}>{note.body}</Text>
      <View style={{flexDirection: 'row'}}>
        <Button
          testID={'update-button'}
          onPress={() => getNote(note.id)}
          title="Update"
          color="#841584"
        />
        <Button
          testID={'delete-button'}
          onPress={() => deleteNote(note.id)}
          title="Delete"
          color="#841584"
        />
      </View>
    </View>
  );
};
export default memo(NotesItem);
