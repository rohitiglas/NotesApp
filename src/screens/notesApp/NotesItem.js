import React, {memo} from 'react';
import {Text, TouchableOpacity, StyleSheet} from 'react-native';
const NotesItem = ({note, getNote, deleteNote}) => {
  return (
    <TouchableOpacity
      testID="note-item"
      onPress={() => getNote(note.id)}
      onLongPress={() => {
        deleteNote(note.id);
      }}
      style={[styles.card, {backgroundColor: '#66CCCC'}]}>
      <Text testID={'title'} numberOfLines={1} style={styles.title}>
        {note.title}
      </Text>
      <Text numberOfLines={6} style={styles.note} testID={'description'}>
        {note.body}
      </Text>
    </TouchableOpacity>
  );
};
export default memo(NotesItem);

const styles = StyleSheet.create({
  card: {
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 5,
    },

    elevation: 10,
    shadowRadius: 5,
    shadowOpacity: 1.0,
    borderRadius: 5,
    margin: 20,
    paddingRight: 20,
    width: 150,
    minHeight: 150,
    color: '#fff',
    backgroundColor: '#000',
  },
  create: {
    fontSize: 11,
    alignSelf: 'flex-end',
    color: '#fff',
    right: -10,
    top: 5,
  },
  title: {
    fontSize: 25,
    fontWeight: 'bold',
    color: '#fff',
    top: 10,
    left: 10,
  },
  category: {
    color: '#FFFBFB',
    fontSize: 10,
    top: 8,
    left: 10,
  },
  note: {
    color: '#fff',
    fontSize: 12,
    top: 10,
    left: 10,
  },
});
