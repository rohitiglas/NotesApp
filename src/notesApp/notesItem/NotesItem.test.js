import 'react-native';
import React from 'react';
import {render, cleanup, fireEvent} from '@testing-library/react-native';
import NotesItem from '../NotesItem';

describe('Notes Item', () => {
  let wrapper;
  const note = {id: '10', item: {title: 'Rohit', body: 'Bansal', id: '10'}};
  beforeEach(() => {
    wrapper = render(
      <NotesItem
        key={note.id}
        note={note.item}
        getNote={jest.fn()}
        toggleModal={jest.fn()}
        deleteNote={jest.fn()}
      />,
    );
  });
  afterEach(() => {
    cleanup();
    wrapper = null;
  });
  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should item', function () {
    const {getByTestId} = wrapper;
    expect(getByTestId('title')).toBeDefined();
    expect(getByTestId('description')).toBeDefined();
    expect(getByTestId('title').props.children).toBe('Rohit');
    expect(getByTestId('description').props.children).toBe('Bansal');
  });
});
