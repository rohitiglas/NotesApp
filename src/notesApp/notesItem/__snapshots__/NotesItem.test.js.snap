// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notes Item renders correctly 1`] = `
<View
  style={
    Object {
      "margin": 10,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 22,
        "fontWeight": "bold",
      }
    }
    testID="title"
  >
    Rohit
  </Text>
  <Text
    testID="description"
  >
    Bansal
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={Object {}}
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="update-button"
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#841584",
              },
            ]
          }
        >
          Update
        </Text>
      </View>
    </View>
    <View
      accessibilityRole="button"
      accessibilityState={Object {}}
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="delete-button"
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#841584",
              },
            ]
          }
        >
          Delete
        </Text>
      </View>
    </View>
  </View>
</View>
`;
